src/agent/multitype_queue/src/multitype_queue.cpp

```diff
@@ -155,47 +135,96 @@ Message MultiTypeQueue::getNext(MessageType type, const std::string moduleName,
     }
     else
     {
-        // TODO: error handling
         LogError("Error didn't find the queue.");
     }
     return result;
 }
 
-boost::asio::awaitable<std::vector<Message>> MultiTypeQueue::getNextNAwaitable(MessageType type,
-                                                                               int messageQuantity,
-                                                                               const std::string moduleName,
-                                                                               const std::string moduleType)
+boost::asio::awaitable<std::vector<Message>>
+MultiTypeQueue::getNextNAwaitable(MessageType type,
+                                  std::variant<const int, const MessageSize> messageQuantity,
+                                  const std::string moduleName,
+                                  const std::string moduleType)
 {
     boost::asio::steady_timer timer(co_await boost::asio::this_coro::executor);
 
     std::vector<Message> result;
     if (m_mapMessageTypeName.contains(type))
     {
-        while (isEmpty(type))
+        if (std::holds_alternative<const int>(messageQuantity))
         {
-            timer.expires_after(std::chrono::milliseconds(DEFAULT_TIMER_IN_MS));
-            co_await timer.async_wait(boost::asio::use_awaitable);
+            // waits for items to be available
+            while (isEmpty(type))
```

Comment:
If we want to keep this case, shouldn't we return the exact number of requested messages unless the timeout expires?

Reply:
that's what it does. That part of the code it's just waiting for messages available in order to avoid a query on an empty table.